src/Functions/filesystem.cpp

```diff
@@ -1,31 +1,40 @@
-#include <Functions/IFunction.h>
-#include <Functions/FunctionFactory.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnVector.h>
 #include <DataTypes/DataTypesNumber.h>
+#include <Disks/IDisk.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/IFunction.h>
 #include <Interpreters/Context.h>
-#include <filesystem>
 #include <Poco/Util/AbstractConfiguration.h>
 
 namespace DB
 {
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int UNKNOWN_DISK;
+}
 namespace
 {
 
 struct FilesystemAvailable
 {
     static constexpr auto name = "filesystemAvailable";
-    static std::uintmax_t get(const std::filesystem::space_info & spaceinfo) { return spaceinfo.available; }
+    static std::uintmax_t get(const DiskPtr & disk) { return disk->getAvailableSpace(); }
 };
 
-struct FilesystemFree
+struct FilesystemUnreserved
 {
-    static constexpr auto name = "filesystemFree";
-    static std::uintmax_t get(const std::filesystem::space_info & spaceinfo) { return spaceinfo.free; }
+    static constexpr auto name = "filesystemUnreserved";
```

Comment:
I'm not sure that we can change the name now, because this function was added a long time ago. Maybe add an alias for old name?

Reply:
But `filesystemFree` is not same as `filesystemUnreserved`, and we can not get free space from `IDisk` interface, so I changed the function to `filesystemUnreserved`.

Reply:
Ok. Then, let's mention it in changelog and mark PR as backward-incompatible.